include/boost/compute/utility/program_cache.hpp

```diff
@@ -82,6 +82,25 @@ class program_cache : boost::noncopyable
         m_cache.clear();
     }
 
+    /// Clears persistent cache by deleting the boost_compute folder
+    void clear_persistent_cache()
+    {
+#ifdef BOOST_COMPUTE_USE_OFFLINE_CACHE
+        // Path delimiter symbol for the current OS.
+        static const std::string delim = boost::filesystem::path("/").make_preferred().string();
+
+        // Path to appdata folder.
```

Comment:
Instead of copying these lines, could you just call `detail::appdata_path()` directly?